### DISCUSSION TITLE
P1P2 and LAN Adapter

### DISCUSSION THREAD

**Discussion by Tardymo** [Arnold-n/P1P2MQTT#51] [General]
I am deciding between LAN and WLAN adapters for my Altherma. WLAN module connects to MMI board, while LAN adapter uses P1-P2 on main board. If I go for the LAN module, will I be able to use P1P2-ESP interface in parallel to LAN module? Is P1P2 bus multi-master?

**Comment by Arnold-n** [comment]
Hi @Tardymo, the P1/P2 bus has only one master; most systems allow only one auxiliary controller (slave), some allow more auxiliary controllers. The P1P2-ESP-interface acts as an auxiliary controller, so in most system the LAN adapter cannot be used. Whether the WLAN module can still be used, and how MMI and P1/P2 are separate or interact, is still unknown.

The P1P2-ESP-interface can always be used in listen-only mode for monitoring purposes.

But if  the P1P2-ESP-interface would be able to control your system (depends on your system of course), would you still need the LAN adapter at all?

**Reply by Tardymo** [reply]
> But if the P1P2-ESP-interface would be able to control your system (depends on your system of course), would you still need the LAN adapter at all?

Probably not. But I have a couple of Air-to-air heatpumps from Daikin as well, which I am able to control through Onecta app. And as far as I know, MMI software is remote upgradeable if there is a LAN/WLAN adapter.
